### PR TITLE
Add a table to describe different use cases for our facades

### PR DESCRIPTION
**Pull request by KEggensperger**
see title ^

### CODE REVIEW

File: doc/usage_recomendation.rst [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+| conditional parameter | no       | no  | yes | no  | yes | no  | yes | no  | yes   |
++-----------------------+----------+-----+-----+-----+-----+-----+-----+-----+-------+
+| instances             | no       | None or CV-folds      | yes                     |
++-----------------------+----------+-----+-----+-----+-----+-----+-----+-----+-------+
+| deterministic         | yes      | can handle both       | can handle both         |
++-----------------------+----------+-----+-----+-----+-----+-----+-----+-----+-------+
+| objective             | any      | e.g. validation loss  | e.g. runtime or quality |
++-----------------------+----------+-----+-----+-----+-----+-----+-----+-----+-------+
+
+Some examples one typical use cases:
```
Comment (mlindauer) [inline doc/usage_recomendation.rst]: one -> of ?

File: doc/usage_recomendation.rst [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+*SMAC4BO*: Vanilla Bayesian Optimization using a *Gaussian Process* and *Expected Improvement*
+   - Optimizing the objective value of Branin and other low dimensional artificial testfunctions
```
Comment (mlindauer) [inline doc/usage_recomendation.rst]: testfunctions -> test functions

File: doc/usage_recomendation.rst [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+   - Finding the best learning rate for training a neuronal network wrt. to RMSE on a validation dataset
```
Comment (mlindauer) [inline doc/usage_recomendation.rst]: neuronal -> neural

File: doc/usage_recomendation.rst [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+*SMAC4HPO*: Bayesian optimization using a *Random Forest*
+  - Finding the optimal choice of machine learning algorithm and its hyperparameters wrt. validation error
+  - Tuning the architecture and training parameters of a neural network wrt. to classification error on a validation dataset
```
Comment (mlindauer) [inline doc/usage_recomendation.rst]: wrt to -> wrt

File: doc/usage_recomendation.rst [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  - Optimize hyperparameters of a SVM wrt. the CV-fold error
+  - Minimize objective values of problems that are noisy and/or yield crashed runs (e.g. due to mem-outs)
+  - Finding the best setting of an RL-agent to minimize training time for a RL problem (or a set of RL problems)
```
Comment (mlindauer) [inline doc/usage_recomendation.rst]: why minimize training time? Time sounds more like AC facade

File: doc/usage_recomendation.rst [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+*SMAC4AC*: Algorithm configuration using a *Random Forest*
+  - Minimizing the average time it takes for a SAT-solver to solve a set of SAT instances
+  - Configuring the CPLEX solver to solve a set of mixed-integer-problems as fast as possible
```
Comment (mlindauer) [inline doc/usage_recomendation.rst]: CPLEX -> MIP

File: doc/usage_recomendation.rst [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+*SMAC* of course itself offers a lot of design choices, some of which are crucial to achieve peak performance. Luckily, often it is enough to distinguish between a few problem classes.
+To make the usage of *SMAC* as easy as possible, we provide several facades designed for these different use cases. Here we give some general recommendations on
+when to use which facade. This recommendation is based on our experience and technical limitations and is by far not intended to be complete:
+
++-----------------------+----------+-----+-----+-----+-----+-----+-----+-----+-------+
+|                       | SMAC4BO  | SMAC4HPO              | SMAC4AC                 |
++=======================+==========+=====+=====+=====+=====+=====+=====+=====+=======+
+| # parameter           | low      | low/medium/high       | low/medium/high         |
++-----------------------+----------+-----+-----+-----+-----+-----+-----+-----+-------+
+| categorical parameter | no       | yes | yes | no  | no  | yes | yes | no  | no    |
```
Comment (mlindauer) [inline doc/usage_recomendation.rst]: yes | yes | no | no --> supported

File: doc/usage_recomendation.rst [hunk 8]
```diff
@@ -0,0 +1,46 @@
+.. _scenario: options.html#scenario
+.. _PCS: options.html#paramcs
+.. _TAE: tae.html
+
+Usage Recommendation
+===========
+*SMAC* of course itself offers a lot of design choices, some of which are crucial to achieve peak performance. Luckily, often it is enough to distinguish between a few problem classes.
```
Comment (mlindauer) [inline doc/usage_recomendation.rst]: enough --> sufficient

File: doc/usage_recomendation.rst [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
++=======================+==========+=====+=====+=====+=====+=====+=====+=====+=======+
+| # parameter           | low      | low/medium/high       | low/medium/high         |
++-----------------------+----------+-----+-----+-----+-----+-----+-----+-----+-------+
+| categorical parameter | no       | yes | yes | no  | no  | yes | yes | no  | no    |
++-----------------------+----------+-----+-----+-----+-----+-----+-----+-----+-------+
+| conditional parameter | no       | no  | yes | no  | yes | no  | yes | no  | yes   |
++-----------------------+----------+-----+-----+-----+-----+-----+-----+-----+-------+
+| instances             | no       | None or CV-folds      | yes                     |
++-----------------------+----------+-----+-----+-----+-----+-----+-----+-----+-------+
+| deterministic         | yes      | can handle both       | can handle both         |
```
Comment (mlindauer) [inline doc/usage_recomendation.rst]: What is "both"?

File: doc/usage_recomendation.rst [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
++-----------------------+----------+-----+-----+-----+-----+-----+-----+-----+-------+
+| objective             | any      | e.g. validation loss  | e.g. runtime or quality |
```
Comment (mlindauer) [inline doc/usage_recomendation.rst]: why any for 4BO? it should not be runtime, right?

File: doc/usage_recomendation.rst (hunk 5 above)
Comment (KEggensperger) [reply doc/usage_recomendation.rst]: Agree, changed it to regret.

File: doc/usage_recomendation.rst (hunk 9 above)
Comment (KEggensperger) [reply doc/usage_recomendation.rst]: I thought of "both" for deterministic/non-deterministic, but now I changed it to
```| stochasticity         | no                   | supported             | supported               |```

File: doc/usage_recomendation.rst (hunk 10 above)
Comment (KEggensperger) [reply doc/usage_recomendation.rst]: Agree, now it says any (except runtime)

File: doc/usage_recomendation.rst [hunk 11]
```diff
@@ -0,0 +1,46 @@
+.. _scenario: options.html#scenario
+.. _PCS: options.html#paramcs
+.. _TAE: tae.html
+
+Usage Recommendation
+===========
```
Comment (mfeurer) [inline doc/usage_recomendation.rst]: Could you please extend the number of equal characters to be of equal length as the header?

File: doc/usage_recomendation.rst [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+*SMAC* of course itself offers a lot of design choices, some of which are crucial to achieve peak performance. Luckily, often it is sufficient to distinguish between a few problem classes.
+To make the usage of *SMAC* as easy as possible, we provide several facades designed for these different use cases. Here we give some general recommendations on
+when to use which facade. This recommendation is based on our experience and technical limitations and is by far not intended to be complete:
```
Comment (mfeurer) [inline doc/usage_recomendation.rst]: In the line above, recommendation is plural, here it's singular.

File: doc/usage_recomendation.rst [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+| instances             | no                   | None or CV-folds      | yes                     |
++-----------------------+----------------------+-----------------------+-------------------------+
+| stochasticity         | no                   | supported             | supported               |
++-----------------------+----------------------+-----------------------+-------------------------+
+| objective             | any (except runtime) | e.g. validation loss  | e.g. runtime or quality |
++-----------------------+----------------------+-----------------------+-------------------------+
+
+Some examples of typical use cases:
+
+*SMAC4BO*: Vanilla Bayesian Optimization using a *Gaussian Process* and *Expected Improvement*
```
Comment (mfeurer) [inline doc/usage_recomendation.rst]: I'm wondering if this is vanilla BO as we integrate the hyperparameters of the Gaussian Process.

**Pull request merged**